bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -564,48 +565,45 @@ public void actionPerformed(final ActionEvent e) {
         }
     };
 
-    private final PropertyChangeListener selectionListener = new PropertyChangeListener() {
-        @Override public void propertyChange(final PropertyChangeEvent evt) {
-            final ISPObsComponent node = getContextTargetObsComp();
-            final TargetEnvironment env = getDataObject().getTargetEnvironment();
-            final Option<SPTarget> targetOpt = TargetSelection.getTargetForNode(env, node);
+    private void setSelectionFromNode() {
+        final ISPObsComponent node = getContextTargetObsComp();
+        final TargetEnvironment env = getDataObject().getTargetEnvironment();
+        final Option<SPTarget> targetOpt = TargetSelection.getTargetForNode(env, node);
+        final Option<IndexedGuideGroup> iggOpt = TargetSelection.getIndexedGuideGroupForNode(env, node);
 
-            // We must be careful here to make sure the _curPos \/ _curGroup property is maintained, i.e. exactly
-            // one of the two must be defined and the other null.
-            if (targetOpt.isDefined()) {
-                // _curGroup = null is handled in function.
-                targetOpt.foreach(target -> manageCurPosIfEnvContainsTarget(target, () -> setSelectionToTarget(target)));
-            } else {
-                final Option<IndexedGuideGroup> iggOpt = TargetSelection.getIndexedGuideGroupForNode(env, node);
-                iggOpt.filter(igg -> env.getGroups().contains(igg.group())).foreach(igg -> {
-                    selectedTarget().foreach(t -> t.deleteWatcher(posWatcher));
-
-                    setSelectionToGroup(igg);
-
-                    _w.guideGroupPanel.setVisible(true);
-                    _w.detailEditor.setVisible(false);
-
-                    // N.B. don't trim, otherwise user can't include space in group name
-                    final String name;
-                    final boolean enabled;
-                    if (selectionIsAutoGroup()) {
-                        // TODO: Possibly change this for different AutomaticGroup types?
-                        name = "Automatic Group";
-                        enabled = false;
-                    } else {
-                        name = igg.group().getName().getOrElse("");
-                        enabled = true;
-                    }
-                    if (!_w.guideGroupName.getValue().equals(name))
-                        _w.guideGroupName.setValue(name);
-                    _w.guideGroupName.setEnabled(enabled);
+        // If a target, process it.
+        targetOpt.foreach(target -> manageCurPosIfEnvContainsTarget(target, () -> setSelectionToTarget(target)));
 
-                    updateUIForGroup();
-                    updateGuideStarAdders();
-                });
+        // If it is a guide group, process it.
+        iggOpt.filter(igg -> env.getGroups().contains(igg.group())).foreach(igg -> {
```

Comment:
What does the `filter` accomplish? Will this suffer from the same issue that two empty groups are considered equal?